### PR TITLE
Use postgres DB while connecting to RDS during restore

### PR DESCRIPTION
**Pull request by PrasadG193**
## Change Overview

During RDS restore, it needs a database to make the connection.
This PR removes overhead of listing all the DBs from the postgres to use one. Instead use default database ie postgres always during restore.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

Tested RDS backup restore using [rds-postgres-dump-blueprint](https://github.com/kanisterio/kanister/blob/master/examples/aws-rds/postgresql/rds-postgres-dump-blueprint.yaml) BP manually as well as by running the integration tests.

### CODE REVIEW

File: pkg/postgres/client.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ import (
 	_ "github.com/lib/pq"
 )
 
+const DefaultConnectDatabase = "postgres"
```
Comment (akankshakumari393) [inline pkg/postgres/client.go:26]:
```suggestion
const defaultConnectDatabase = "postgres"
```

File: pkg/function/restore_rds_snapshot.go [hunk 2]
```diff
@@ -237,7 +235,7 @@ func postgresRestoreCommand(pgHost, username, password string, dbList []string,
 		fmt.Sprintf(`
 		export PGHOST=%s
 		kando location pull --profile '%s' --path "%s" - | gunzip -c -f |%s psql -q -U "${PGUSER}" %s
-		`, pgHost, profile, fmt.Sprintf("%s/%s", backupArtifactPrefix, backupID), replaceCommand, dbList[0]),
+		`, pgHost, profile, fmt.Sprintf("%s/%s", backupArtifactPrefix, backupID), replaceCommand, postgres.DefaultConnectDatabase),
```
Comment (akankshakumari393) [inline pkg/function/restore_rds_snapshot.go:238]:
```suggestion
		`, pgHost, profile, fmt.Sprintf("%s/%s", backupArtifactPrefix, backupID), replaceCommand, postgres.defaultConnectDatabase),
```

**Review by akankshakumari393 — APPROVED** [review]
LGTM

File: pkg/function/restore_rds_snapshot.go (hunk 2 above)
Comment (PrasadG193) [reply pkg/function/restore_rds_snapshot.go:238]: @akankshakumari393 the const is declared in another package. To import and use the const from different package, it needs to be exported.

File: pkg/postgres/client.go (hunk 1 above)
Comment (PrasadG193) [reply pkg/postgres/client.go:26]: We need to export the const as it is being imported and used in another package.

**Review by viveksinghggits — APPROVED** [review]

File: pkg/function/export_rds_snapshot_location.go [hunk 3]
```diff
@@ -250,48 +250,65 @@ func execDumpCommand(ctx context.Context, dbEngine RDSDBEngine, action RDSAction
...
-			command, err := postgresRestoreCommand(dbEndpoint, username, password, dbList, backupPrefix, backupID, profileJson, dbEngineVersion)
+			command, err := postgresRestoreCommand(dbEndpoint, username, password, backupPrefix, backupID, profileJson, dbEngineVersion)
 			return command, postgresToolsImage, err
 		}
 	}
 	return nil, "", errors.New("Invalid RDSDBEngine or RDSAction")
 }
 
+func findDBList(ctx context.Context, dbEndpoint, username, password string) ([]string, error) {
+	// Find list of dbs
```
Comment (PrasadG193) [inline pkg/function/export_rds_snapshot_location.go:294]:
```suggestion
```
Remove misleading comment

**Pull request merged**